File: Buddy/buddy-flutter/lib/features/home/components/lcd_card.dart
import 'package:flutter/material.dart';

import '../../../design/theme.dart';
import '../../../models/pet.dart';

class LCDCard extends StatelessWidget {
  final Pet pet;
  const LCDCard({super.key, required this.pet});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 20),
      child: Container(
        height: 170,
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: BuddyTheme.lcdOuter,
          borderRadius: BorderRadius.circular(20),
          border: Border.all(color: BuddyTheme.lcdStroke.withValues(alpha: 0.4), width: 2),
        ),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            _avatar(),
            const SizedBox(width: 12),
            Container(width: 2, color: BuddyTheme.lcdStroke),
            const SizedBox(width: 12),
            Expanded(child: _stats()),
          ],
        ),
      ),
    );
  }

  Widget _avatar() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Container(
          width: 80, height: 70,
          decoration: BoxDecoration(
            color: Colors.white,
            borderRadius: BorderRadius.circular(6),
            border: Border.all(color: BuddyTheme.lcdStroke, width: 2),
          ),
          alignment: Alignment.center,
          child: Text(pet.character.emoji, style: const TextStyle(fontSize: 36)),
        ),
        const SizedBox(height: 4),
        Row(children: [
          Text(pet.name, style: BuddyTheme.pixel(size: 18, weight: FontWeight.bold, color: BuddyTheme.lcdInk)),
          const SizedBox(width: 4),
          const Text('✏', style: TextStyle(fontSize: 12)),
        ]),
      ],
    );
  }

  Widget _stats() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
      children: [
        _row('Age', '${pet.ageInDays} Days'),
        _row('Mood', _hearts(pet.moodLevel, 3)),
        _row('Satiety', _dots(pet.satietyLevel, 2)),
      ],
    );
  }

  Widget _row(String label, String value) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceBetween,
      children: [
        Text(label, style: BuddyTheme.pixel(size: 16, color: BuddyTheme.lcdInk)),
        Text(
          value,
          style: BuddyTheme.pixel(
            size: 16,
            color: label == 'Mood' ? const Color(0xFFE04878) : BuddyTheme.lcdInk,
          ),
        ),
      ],
    );
  }

  String _hearts(int filled, int total) =>
      List.generate(total, (i) => i < filled ? '♥' : '♡').join(' ');

  String _dots(int filled, int total) =>
      List.generate(total, (i) => i < filled ? '●' : '○').join(' ');
}

File: Buddy/buddy-ios/Sources/buddy/Features/Home/Components/LCDCard.swift
import SwiftUI

struct LCDCard: View {
    let pet: Pet

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 20)
                .fill(Theme.lcdOuter)
                .overlay(
                    RoundedRectangle(cornerRadius: 20)
                        .stroke(Theme.lcdStroke.opacity(0.4), lineWidth: 2)
                )

            HStack(spacing: 12) {
                avatar
                Rectangle()
                    .fill(Theme.lcdStroke)
                    .frame(width: 2)
                stats
            }
            .padding(16)
        }
        .frame(height: 170)
        .padding(.horizontal, 20)
    }

    private var avatar: some View {
        VStack(alignment: .leading, spacing: 4) {
            ZStack {
                RoundedRectangle(cornerRadius: 6)
                    .fill(.white)
                    .overlay(
                        RoundedRectangle(cornerRadius: 6)
                            .stroke(Theme.lcdStroke, lineWidth: 2)
                    )
                Text("🐱")
                    .font(.system(size: 36))
            }
            .frame(width: 80, height: 70)
            HStack(spacing: 4) {
                Text(pet.name)
                    .font(.custom(Theme.pixelMono, size: 18).weight(.bold))
                    .foregroundStyle(Theme.lcdInk)
                Text("✏")
                    .font(.system(size: 12))
                    .foregroundStyle(Theme.lcdInk)
            }
        }
    }

    private var stats: some View {
        VStack(alignment: .leading, spacing: 12) {
            row(label: "Age", value: "\(pet.ageInDays) Days")
            row(label: "Mood", value: hearts(filled: pet.moodLevel, total: 3))
            row(label: "Satiety", value: dots(filled: pet.satietyLevel, total: 2))
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private func row(label: String, value: String) -> some View {
        HStack {
            Text(label)
                .font(.custom(Theme.pixelMono, size: 16))
                .foregroundStyle(Theme.lcdInk)
            Spacer()
            Text(value)
                .font(.custom(Theme.pixelMono, size: 16))
                .foregroundStyle(label == "Mood" ? Color(red: 0.878, green: 0.282, blue: 0.471) : Theme.lcdInk)
        }
    }

    private func hearts(filled: Int, total: Int) -> String {
        (0..<total).map { $0 < filled ? "♥" : "♡" }.joined(separator: " ")
    }

    private func dots(filled: Int, total: Int) -> String {
        (0..<total).map { $0 < filled ? "●" : "○" }.joined(separator: " ")
    }
}
